precompiles/staking/approve_test.go

```diff
@@ -216,37 +220,34 @@ func (s *PrecompileTestSuite) TestApprove() {
 		{
 			"success - Authorization should only be created for validators that are not jailed",
 			func(_ *vm.Contract) []interface{} {
-				// Commit block (otherwise test logic will not be executed correctly, i.e. somehow unbonding does not take effect)
 				var err error
-				s.ctx, err = evmosutil.Commit(s.ctx, s.app, time.Second, nil)
-				s.Require().NoError(err, "failed to commit block")
-
 				// Jail a validator
-				s.app.StakingKeeper.Jail(s.ctx, sdk.ConsAddress(s.validators[0].GetOperator()))
+				valAddr, err = sdk.ValAddressFromBech32(s.network.GetValidators()[0].GetOperator())
+				s.Require().NoError(err)
+				err = s.network.App.StakingKeeper.Jail(ctx, sdk.ConsAddress(valAddr))
+				s.Require().NoError(err, "failed to jail a validator")
 
 				// When a delegator redelegates/undelegates from a validator, the validator
 				// switches to Unbonding status.
 				// Thus, validators with this status should be considered for the authorization
 
 				// Unbond another validator
-				amount, err := s.app.StakingKeeper.Unbond(s.ctx, s.address.Bytes(), sdk.ValAddress(s.validators[1].GetOperator()), math.LegacyOneDec())
+				valAddr1, err := sdk.ValAddressFromBech32(s.network.GetValidators()[1].GetOperator())
+				s.Require().NoError(err)
+				amount, err := s.network.App.StakingKeeper.Unbond(ctx, s.keyring.GetAddr(0).Bytes(), valAddr1, math.LegacyOneDec())
 				s.Require().NoError(err, "expected no error unbonding validator")
 				s.Require().Equal(math.NewInt(1e18), amount, "expected different amount of tokens to be unbonded")
 
-				// Commit block and update time to one year later
-				s.ctx, err = evmosutil.Commit(s.ctx, s.app, time.Hour*24*365, nil)
-				s.Require().NoError(err, "failed to commit block")
-
 				return []interface{}{
-					s.address,
+					s.keyring.GetAddr(0),
 					big.NewInt(1e18),
 					[]string{staking.DelegateMsg},
 				}
 			},
 			func(data []byte, inputArgs []interface{}) {
 				s.Require().Equal(data, cmn.TrueValue)
 
-				authz, expirationTime := s.CheckAuthorization(staking.DelegateAuthz, s.address, s.address)
+				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(0), s.keyring.GetAddr(0))
```

Comment:
this anti-pattern of creating self-approvals unfortunately is present in every single test it seems..


```suggestion
				authz, expirationTime := CheckAuthorizationWithContext(ctx, s.network.App.AuthzKeeper, staking.DelegateAuthz, s.keyring.GetAddr(granteeIdx), s.keyring.GetAddr(granterIdx))
```